### PR TITLE
feat: add VALI-based (GPU) video reader

### PR DESCRIPTION
**Pull request by egorchakov**
Tested with one of VALI's own files: https://github.com/RomanArzumanyan/VALI/blob/main/tests/data/test.mp4

NOTE: seeking seems problematic (https://github.com/RomanArzumanyan/VALI/pull/111), might need to hold off on merging

### CODE REVIEW

**Comment by sandhawalia** [comment]
Shall we update [yaak dataset config](https://github.com/yaak-ai/rbyte/blob/main/examples/config_templates/dataset/yaak.yaml) and verify if the seek issue persists?

**Comment by egorchakov** [comment]
We can't actually use it on our videos since they're all VFR. I tried converting one to CFR, and the seek issue was definitely there. Took me a while to trace it back though.

File: src/rbyte/io/frame/video/ffmpeg_reader.py [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
 from functools import partial
-from os import PathLike
 from pathlib import Path
 from typing import override
 
 import torch
 import video_reader as vr
 from jaxtyping import UInt8
-from pydantic import NonNegativeInt, validate_call
+from pydantic import FilePath, NonNegativeInt, validate_call
```
Comment (sandhawalia) [inline src/rbyte/io/frame/video/ffmpeg_reader.py:9]: ❤️

**Review by sandhawalia — APPROVED** [review]
LGTM pending seek issue fix

**Pull request merged**